src/engine/Actor.ts

```diff
@@ -471,6 +471,8 @@ export class Actor extends Class implements IActionable, IEvented {
       }
    }
 
+   public on(eventName: Events.collision, handler: (event?: CollisionEvent) => void): void;
```

Comment:
Let's rename this to precollision and update references (deprecate `collision`)

Reply:
Ah in actor `update` and `postupdate` are equivalent with this todo

```typescript
      // TODO: Obsolete `update` event on Actor
      this.eventDispatcher.emit('update', new PostUpdateEvent(engine, delta, this));
      this.emit('postupdate', new PostUpdateEvent(engine, delta, this));

```